tests/Unit/Evolution/DiscontinuousGalerkin/Limiters/Test_SimpleWenoImpl.cpp

```diff
@@ -0,0 +1,553 @@
+    get<2>(get<VectorTag<3>>(vars)) = y + square(x) * z;
+    return vars;
+  };
+  const auto make_lower_eta_vars =
+      [](const tnsr::I<DataVector, 3, Frame::Logical>& coords,
+         const double eta_offset = 0.0) noexcept {
+    const auto& x = get<0>(coords);
+    const auto y = get<1>(coords) + eta_offset;
+    const auto& z = get<2>(coords);
+    Variables<tmpl::list<VectorTag<3>>> vars(x.size());
+    get<0>(get<VectorTag<3>>(vars)) = 0.4 * x * y * z + square(z);
+    get<1>(get<VectorTag<3>>(vars)) = -0.1 * y + z;
+    get<2>(get<VectorTag<3>>(vars)) = -square(z);
+    return vars;
+  };
+  const auto make_upper_eta_vars =
+      [](const tnsr::I<DataVector, 3, Frame::Logical>& coords,
+         const double eta_offset = 0.0) noexcept {
+    const auto& x = get<0>(coords);
+    const auto y = get<1>(coords) + eta_offset;
+    const auto& z = get<2>(coords);
+    Variables<tmpl::list<VectorTag<3>>> vars(x.size());
+    get<0>(get<VectorTag<3>>(vars)) = 0.4 * x * y * z + square(z);
+    get<1>(get<VectorTag<3>>(vars)) = z + 0.4 * x * cube(z);
+    get<2>(get<VectorTag<3>>(vars)) = y * z + square(y) + cube(z);
+    return vars;
+  };
+  const auto make_lower_zeta_vars =
+      [](const tnsr::I<DataVector, 3, Frame::Logical>& coords,
+         const double zeta_offset = 0.0) noexcept {
+    const auto& x = get<0>(coords);
+    const auto& y = get<1>(coords);
+    const auto z = get<2>(coords) + zeta_offset;
+    Variables<tmpl::list<VectorTag<3>>> vars(x.size());
+    get<0>(get<VectorTag<3>>(vars)) = 0.4 * x * y * z + square(z);
+    get<1>(get<VectorTag<3>>(vars)) = 0.9 * z - 2. * x * z;
+    get<2>(get<VectorTag<3>>(vars)) = y + cube(z);
+    return vars;
+  };
+  const auto make_upper_zeta_vars =
+      [](const tnsr::I<DataVector, 3, Frame::Logical>& coords,
+         const double zeta_offset = 0.0) noexcept {
+    const auto& x = get<0>(coords);
+    const auto& y = get<1>(coords);
+    const auto z = get<2>(coords) + zeta_offset;
+    Variables<tmpl::list<VectorTag<3>>> vars(x.size());
+    get<0>(get<VectorTag<3>>(vars)) = 0.4 * x * y * z + square(z);
+    get<1>(get<VectorTag<3>>(vars)) = 1.3 * square(y) * square(z);
+    get<2>(get<VectorTag<3>>(vars)) = -x * y * z + square(y);
+    return vars;
+  };
+
+  const auto local_data = make_center_tensor(logical_coords);
+  VariablesMap<3> neighbor_vars{};
+  VariablesMap<3> neighbor_modified_vars{};
+
+  const auto shift_vars_to_local_means = [&mesh, &local_data ](
+      const Variables<tmpl::list<VectorTag<3>>>& input) noexcept {
+    auto result = input;
+    auto& v = get<VectorTag<3>>(result);
+    get<0>(v) +=
+        mean_value(get<0>(local_data), mesh) - mean_value(get<0>(v), mesh);
+    get<1>(v) +=
+        mean_value(get<1>(local_data), mesh) - mean_value(get<1>(v), mesh);
+    get<2>(v) +=
+        mean_value(get<2>(local_data), mesh) - mean_value(get<2>(v), mesh);
+    return result;
+  };
+
+  const auto lower_xi =
```

Comment:
Is it possible to factor all these blocks into a single function that you pass the `std::pair` and the `make_lower_BLAH_vars` to?